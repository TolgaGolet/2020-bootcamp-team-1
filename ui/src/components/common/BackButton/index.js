import React from "react";
import Button from "react-bootstrap/Button";
import { useHistory } from "react-router-dom";

const BackButton = () => {
  const { push } = useHistory();
  const goBack = () => {
    push("/");
  };

  return (
    <Button className="mb-5" size="lg" onClick={goBack}>
      Go back to home
    </Button>
  );
};

export default BackButton;
