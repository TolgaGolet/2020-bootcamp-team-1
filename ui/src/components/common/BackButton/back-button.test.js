import React from "react";
import { shallow } from "enzyme";
import BackButton from "./";

const setup = (props = {}) => {
  const component = shallow(<BackButton {...props} />);
  return component;
};

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: jest.fn()
  })
}));

describe("BackButton Component", () => {
  it("Renders correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders without prop and default `Go back to home` text correctly", () => {
    const wrapper = setup();
    const text = wrapper.text();
    expect(text).toBe("Go back to home");
  });
  
});
