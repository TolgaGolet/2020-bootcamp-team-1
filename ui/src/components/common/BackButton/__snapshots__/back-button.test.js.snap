// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Component Renders correctly 1`] = `ShallowWrapper {}`;
